src/hssm/hssm.py

```diff
@@ -191,148 +191,50 @@ def __init__(
         self.data = data
         self._inference_obj = None
         self.hierarchical = hierarchical and "participant_id" in data.columns
-        self.has_lapse = p_outlier is not None and p_outlier != 0
 
-        if loglik_kind is None:
-            if model not in default_model_config:
-                raise ValueError(
-                    "When using a custom model, please provide a `loglik_kind.`"
-                )
-            # Setting loglik_kind to be the first of analytical or
-            # approx_differentiable
-            for kind in ["analytical", "approx_differentiable", "blackbox"]:
-                model = cast(SupportedModels, model)
-                if kind in default_model_config[model]:
-                    kind = cast(LoglikKind, kind)
-                    loglik_kind = kind
-                    break
-            if loglik_kind is None:
-                raise ValueError(
-                    "No default model_config is found. Please provide a `loglik_kind."
-                )
-        else:
-            if loglik_kind not in [
-                "analytical",
-                "approx_differentiable",
-                "blackbox",
-            ]:
-                raise ValueError(
-                    "'loglike_kind', when provided, must be one of "
-                    + '"analytical", "approx_differentiable", "blackbox".'
-                )
-
-        self.loglik_kind = loglik_kind
-        self.model_name = model
-
-        # Check if model has default config
-        if _model_has_default(self.model_name, self.loglik_kind):
-            model = cast(SupportedModels, model)
-            default_config = default_model_config[model][loglik_kind]
-
-            self.model_config = (
-                deepcopy(default_config)
-                if model_config is None
-                else merge_dicts(default_config, model_config)
+        # Construct a model_config from defaults
+        self.model_config = Config.from_defaults(model, loglik_kind)
+        # Update defaults with user-provided config, if any
+        if model_config is not None:
```

Comment:
why would model_config end up being None at this point?
Config.from_defaults can return None? 
But if it would, then you wouldn't even enter the if clause?!

Reply:
Ah, `model_config` is user supplied. `self.model_config` is a different object at this point